src/shogun/evaluation/SplittingStrategy.h

```diff
@@ -58,33 +58,29 @@ class CSplittingStrategy: public CSGObject
 	 * @param subset_idx subset index of the to be generated vector indices
 	 * @return newly created vector of subset indices of the specified
 	 * subset is written here.
-	 *
-	 * Error if there are no index sets
 	 */
-	SGVector<index_t> generate_subset_indices(index_t subset_idx);
+	const SGVector<index_t> validation(const index_t subset_idx) const;
```

Comment:
not sure why we want the return type to be const?

Reply:
I've changed this according to the Issue description. I think this shouldn't be a problem, given the possibility to return it to const or non-const variables. Should I remove it?

Reply:
I think it should be ok